src/subhalo_unbind.cpp

```diff
@@ -442,8 +442,15 @@ void Subhalo_t::Unbind(const Snapshot_t &epoch)
         }
         // update particle list
         sort(Elist.begin(), Elist.begin() + Nbound, CompEnergy); // sort the self-bound part
-        if (RefineMostboundParticle && Nbound > MaxSampleSize)   // refine most-bound particle, not necessary usually..
-          RefineBindingEnergyOrder(ESnap, MaxSampleSize, tree, RefPos, RefVel);
+
+        /* We need to refine the most bound particle, as subsampling large subhaloes will lead to 
+         * incorrect ordering of binding energies. Hence, the most bound particle before this step
+         * may not be the true most bound particle. */
+        if (RefineMostboundParticle && Nbound > MaxSampleSize)
+        {
+          int Nrefine = max(MaxSampleSize, static_cast<HBTInt>(HBTConfig.BoundFractionCentreRefinement * Nbound));
```

Comment:
@robjmcgibbon better to change this from an `int` to an `HBTInt`?

Reply:
I can do it, btw. Just double checking

Reply:
I will change the dtype, as `RefineBindingEnergyOrder` has HBTInt as the dtype for `Nrefine`.